EIPS/eip-2935.md

```diff
@@ -12,43 +12,44 @@ created: 2020-09-03
 
 ## Abstract
 
-Store historical block hashes in a contract, and modify the `BLOCKHASH (0x40)` opcode to read this contract.
+Store last 256 historical block hashes in a contract, and modify the `BLOCKHASH (0x40)` opcode to read and serve from this contract storage.
 
 ## Motivation
 
-There is increasingly a desire to remove the need for most clients to store history older than some relatively short duration (often between 1 week and 1 year) to save disk space. This requires some form of layer-2 network to help clients access historical information. These protocols can be made much simpler if blocks contained a quick Merkle path to historical blocks.
+Currently BLOCKHASH opcode accesses history to resolve hash of the block number in EVM. However a more stateless client friendly way is to maintain and serve these hashes from state.
 
-Additional secondary motivations include:
+Although this is possible even in Merkle trie state, but Verkle trie state further allows bundling the BLOCKHASH witnesses (along with other witnesses) in an efficient manner making it worthwhile to have these in state.
 
-* The protocol can be used to make more secure efficient light clients with flyclient-like technology (while the "optimal" flyclient protocol is fairly complex, large security gains over the status quo (trusted "canonical hash trees") can be made cheaply)
-* Improving cleanness of the protocol, as the BLOCKHASH opcode would then access state and not history.
+A side benefit of this approach could be that it allows building/validating proofs related to last 256 ancestors directly against the current state.
 
 ## Specification
 
 | Parameter | Value |
 | - | - |
 | `FORK_TIMESTAMP` | TBD |
 | `HISTORY_STORAGE_ADDRESS` | `0xfffffffffffffffffffffffffffffffffffffffe`|
-| `MIN_HISTORY_SERVE_WINDOW` | `256` |
-| `MAX_HISTORY_SERVE_WINDOW` | `2**256` |
+| `HISTORY_SERVE_WINDOW` | `256` |
+
+This EIP specifies for storing last `HISTORY_SERVE_WINDOW` block hashes in a ring buffer storage of `HISTORY_SERVE_WINDOW` length.
+
 
 At the start of processing any block where `block.timestamp >= FORK_TIMESTAMP` (ie. before processing any transactions), update the history in the following way:
 
 ```python
 def process_block_hash_history(block :Block, state: State):
     if block.timestamp >= FORK_TIMESTAMP:
```

Comment:
```suggestion
    if block.timestamp >= FORK_TIMESTAMP and block.number > 0:
```

Reply:
ummm  isn't this a bit redundant? genesis block anyway doesn't undergo processing